### PR TITLE
Compiles with Visual Studio 2019

### PR DESCRIPTION
**Pull request by russell-taylor**
This includes the changes I needed to make to get the code to compile against the PDCurses library on WIndows 10 using Visual Studio 2019.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,14 @@ set(CMAKE_CXX_STANDARD_REQUIRED on)
 
 project(umoria)
 
+#
+# Get around the fact that Visual Studio doesn't have ssize_t
+# defined but uses SSIZE_T instead.
+#
+if(MSVC)
+    add_definitions(-Dssize_t=SSIZE_T)
+endif(MSVC)
```
Comment (mrcook) [inline CMakeLists.txt:35]: Does this need to be here at the top of the config? Is it not possible to have it at the bottom of the file? Maybe here: https://github.com/dungeons-of-moria/umoria/blob/master/CMakeLists.txt#L240

**Review by mrcook — COMMENTED** [review]
Hi Russell,

Many thanks for the PR. Other than the one small comment I'd be happy to merge this.

One question: this changes makes it work with PDCurses out of the box?

I was recently wondering what it might take to get umoria compiling with PDCurses so that the mingw dependency could be removed. If this is it, then I might well have to spend some time on my Win10 VM to get that toolchain setup.

File: CMakeLists.txt (hunk 1 above)
Comment (russell-taylor) [reply CMakeLists.txt]: I tend to put those at the top of the file to make it clear that it applies to all things that are added, but it looks like it works just as well down below.  I've changed this in a commit on that branch.

With this, it compiles out of the box; you have to point CURSES_CURSES_LIBRARY at the pdcurses.lib and CURSES_INCLUDE_PATH one directory above where curses.h lives.

To run it, you need to copy the scores.dat file into the location of the executable (which on Windows is a subdirectory under the build directory named based on the build).

Comment (russell-taylor) [reply CMakeLists.txt]: Also the data directory needs to be copied there.

Comment (russell-taylor) [reply CMakeLists.txt]: Thank you for keeping umoria alive!  It was a blast from the past to get to play it again.

Comment (mrcook) [reply CMakeLists.txt]: Great, thanks for the extra info. I will definitely take a look at building the Windows versions with PDCurse.

**Pull request merged**